domain/user/service/service.go

```diff
@@ -233,47 +319,44 @@ func (s *Service) RemoveUser(ctx context.Context, name string) error {
 // - internal/auth.ErrPasswordNotValid: If the password supplied is not valid.
 func (s *Service) SetPassword(
 	ctx context.Context,
-	name string,
+	uuid user.UUID,
 	password auth.Password,
 ) error {
 	defer password.Destroy()
-	if err := ValidateUsername(name); err != nil {
-		return fmt.Errorf("username %q: %w", name, err)
+	if err := uuid.Validate(); err != nil {
+		return errors.Annotatef(err, "validate uuid %q", uuid)
 	}
 
 	salt, err := auth.NewSalt()
 	if err != nil {
-		return fmt.Errorf("setting password for user %q, generating password salt: %w", name, err)
+		return errors.Annotatef(err, "generate password salt for user with uuid %q", uuid)
 	}
 
 	pwHash, err := auth.HashPassword(password, salt)
 	if err != nil {
-		return fmt.Errorf("setting password for user %q, hashing password: %w", name, err)
+		return errors.Annotatef(err, "hash password for user with uuid %q", uuid)
 	}
 
-	if err = s.st.SetPasswordHash(ctx, name, pwHash, salt); err != nil {
-		return fmt.Errorf("setting password for user %q: %w", name, err)
+	if err = s.st.SetPasswordHash(ctx, uuid, pwHash, salt); err != nil {
+		return errors.Annotatef(err, "set password for user with uuid %q", uuid)
 	}
 	return nil
 }
 
 // ResetPassword will remove any active passwords for a user and generate a new
 // activation key for the user to use to set a new password.
-// The following error types are possible from this function:
-// - usererrors.UsernameNotValid: When the username supplied is not valid.
-// - usererrors.NotFound: If no user by the given name exists.
-func (s *Service) ResetPassword(ctx context.Context, name string) ([]byte, error) {
-	if err := ValidateUsername(name); err != nil {
-		return nil, fmt.Errorf("username %q: %w", name, err)
+func (s *Service) ResetPassword(ctx context.Context, uuid user.UUID) ([]byte, error) {
+	if err := uuid.Validate(); err != nil {
+		return nil, errors.Annotatef(err, "validate uuid %q", uuid)
 	}
 
 	activationKey, err := generateActivationKey()
 	if err != nil {
-		return nil, fmt.Errorf("generating activation key for user %q: %w", name, err)
+		return nil, errors.Annotatef(err, "generate activation key for user with uuid %q", uuid)
```

Comment:
Annotate with [present participle](https://www.wordnik.com/words/present%20participle) rather than the verb - "generating" in this case.